lib/blank_video_generator.ex

```diff
@@ -5,69 +5,71 @@ defmodule Membrane.BlankVideoGenerator do
 
   use Membrane.Source
 
-  alias Membrane.Caps.Matcher
   alias Membrane.RawVideo
   alias Membrane.{Buffer, Time}
 
-  @supported_caps {RawVideo, pixel_format: Matcher.one_of([:I420, :I422]), aligned: true}
+  # @supported_stream_format %RawVideo{pixel_format: pixel_format, aligned: true} when pixel_format in [:I420, :I422]
 
   def_options duration: [
-                type: :integer,
                 spec: Time.t(),
                 description: "Duration of the output"
               ],
-              caps: [
-                type: :struct,
+              stream_format: [
                 spec: RawVideo.t(),
                 description: "Video format of the output"
               ]
 
   def_output_pad :output,
-    caps: @supported_caps,
+    accepted_format:
+      %RawVideo{pixel_format: pixel_format, aligned: true} when pixel_format in [:I420, :I422],
     mode: :pull,
     availability: :always
 
   @impl true
-  def handle_init(opts) do
+  def handle_init(_context, opts) do
     cond do
-      !caps_supported?(opts.caps) ->
+      !stream_format_supported?(opts.stream_format) ->
```

Comment:
If operating on booleans, `not` is preferred
```suggestion
      not stream_format_supported?(opts.stream_format) ->
```